node/src/components/linear_chain_sync.rs

```diff
@@ -206,7 +205,11 @@ impl<I: Clone + PartialEq + 'static> LinearChainSync<I> {
         match &mut self.state {
             State::None | State::Done => {}
             State::SyncingTrustedHash { linear_chain, .. } => linear_chain.push(block_header),
-            State::SyncingDescendants { linear_chain, .. } => linear_chain.push_back(block_header),
+            State::SyncingDescendants {
+                linear_chain_block, ..
+            } => {
+                let _ = linear_chain_block.replace(block_header);
```

Comment:
Why not just:
```rust
*linear_chain_block = Some(block_header);
```

Reply:
I thought that I need `replace` in order to mutate `self.state` properly.

Reply:
Done.